test/k8s/simple_test.go

```diff
@@ -103,5 +118,87 @@ data:
 			}()
 		}
 		wg.Wait()
+
+		AddReportEntry("duration", time.Now().Sub(start))
+	})
+
+	It("should deploy mesh wide policy", func() {
+		deliveryCount, err := XdsDeliveryCount(promClient)
+		Expect(err).ToNot(HaveOccurred())
+
+		policy := `
+apiVersion: kuma.io/v1alpha1
+kind: MeshRateLimit
+metadata:
+  name: mesh-rate-limit
+  namespace: kong-mesh-system
+spec:
+  targetRef:
+    kind: Mesh
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+        local:
+          http:
+            requestRate:
+              num: 10000
+              interval: 1s
+            onRateLimit:
+              status: 429
+`
+		Expect(cluster.Install(YamlK8s(policy))).To(Succeed())
+		start := time.Now()
+
+		Eventually(func(g Gomega) {
+			newDeliveryCount, err := XdsDeliveryCount(promClient)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(newDeliveryCount - deliveryCount).To(Equal(numServices))
+		}, "60s", "1s").Should(Succeed())
+		AddReportEntry("duration", time.Now().Sub(start))
+	})
+
+	Context("scaling", func() {
+		scale := func(replicas int) {
+			err := k8s.RunKubectlE(
+				cluster.GetTesting(),
+				cluster.GetKubectlOptions(TestNamespace),
+				"scale", "statefulset", "srv-000", fmt.Sprintf("--replicas=%d", replicas),
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = cluster.Install(WaitNumPods(TestNamespace, replicas, "srv-000"))
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		It("should scale up a service", func() {
+			start := time.Now()
+			scale(2)
+			// there is no straightforward way to check if all Envoys received the config with the new endpoint, therefore we need to rely on stabilization sleep
```

Comment:
Why do we consider "all Envoys received the config" as an assertion for the test? I think scale is successful when all instances of `srv-000` are ready

Reply:
I think the idea is to check whether all consumers of srv-000 have the new endpoint. I'll cover this assertion in an additional PR.

Reply:
@lobkovilya we can discuss this in the next PR where this is actually asserted, sound good?